src/Features/Core/Portable/ConvertAnonymousTypeToTuple/AbstractConvertAnonymousTypeToTupleCodeFixProvider.cs

```diff
@@ -0,0 +1,141 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Immutable;
+using System.Diagnostics;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.CodeActions;
+using Microsoft.CodeAnalysis.CodeFixes;
+using Microsoft.CodeAnalysis.Diagnostics;
+using Microsoft.CodeAnalysis.Editing;
+using Microsoft.CodeAnalysis.Formatting;
+using Microsoft.CodeAnalysis.LanguageServices;
+using Microsoft.CodeAnalysis.Shared.Extensions;
+using Roslyn.Utilities;
+
+namespace Microsoft.CodeAnalysis.ConvertAnonymousTypeToTuple
+{
+    internal abstract class AbstractConvertAnonymousTypeToTupleCodeFixProvider<
+        TExpressionSyntax,
+        TTupleExpressionSyntax,
+        TAnonymousObjectCreationExpressionSyntax> 
+        : SyntaxEditorBasedCodeFixProvider
+        where TExpressionSyntax : SyntaxNode
+        where TTupleExpressionSyntax : TExpressionSyntax
+        where TAnonymousObjectCreationExpressionSyntax : TExpressionSyntax
+    {
+        protected abstract TTupleExpressionSyntax ConvertToTuple(TAnonymousObjectCreationExpressionSyntax anonCreation);
+
+        public override ImmutableArray<string> FixableDiagnosticIds { get; } =
+            ImmutableArray.Create(IDEDiagnosticIds.ConvertAnonymousTypeToTupleDiagnosticId);
+
+        public override Task RegisterCodeFixesAsync(CodeFixContext context)
+        {
+            context.RegisterCodeFix(
+                new MyCodeAction(c => FixAllWithEditorAsync(context.Document,
+                    e => FixInCurrentMember(context.Document, e, context.Diagnostics[0], c), c)),
+                context.Diagnostics);
+
+            return SpecializedTasks.EmptyTask;
```

Comment:
>EmptyTask [](start = 36, length = 9)

I'm not sure why we have this static value. Task.CompletedTask is shorter and seems clearer.

Reply:
I think this predates Task.CompletedTask.  And SpecializedTasks.EmptyTask just defers to Task.CopmletedTask.  I'm happy to make another PR that removes the old form and updates everyone.